### ISSUE TITLE
write() spec is not specific enough to make the cursor usable

### ISSUE THREAD

**Issue by hills** [WICG/file-system-access#336]
Section 2.6.1: The write() method

These two variations:

* ```write({ type: "write", data: data })``` is described as "Writes the content of data into the file associated with stream at the current file cursor offset."

* ```write({ type: "write", position: position, data: data })``` is described as "Writes the content of data into the file associated with stream at position bytes from the top of the file. Also updates the current file cursor offset to the end of the written data."

Leaves the following ambiguous:

* It's left unspecified whether the relative write (first one) advances the cursor -- it probably should. Without spec'ing this, it means the cursor cannot be used.
* It's unspecified (but very important) whether the cursor is advanced synchronously on invocation (preferred) or on completion of the promise. The former being much easier to work, and the latter would make it impossible to do concurrent writes.

Also, it seems counter-intuitive that` writing to an absolute position adjusts the cursor; through the kind of code that writes to absolute position is probably ignoring the cursor.

**Comment by mkruisselbrink** [comment]
I agree that we should make the non-normative notes as clear and unambiguous as possible. Fortunately at least the normative language in the spec is unambiguous about these things; write() queues up a chunk to the stream's underlying sink, which then processes each chunk in order, with every write operation updating the current cursor. So while not exactly synchronous, since every operation on the stream goes through the same queuing mechanism, the effect is still that every new write call will have all the effects of previous write calls already applied, including updating the cursor position, by the time the later write call executes.
